### PR TITLE
[Merged by Bors] - chore: make List.mem_split an alias of List.append_of_mem

### PR DESCRIPTION
**Pull request by dwrensha**
`List.mem_split` duplicates `List.append_of_mem` from `Std`:
https://github.com/leanprover/std4/blob/a756b7d643ae5dcd7bf314e99f8e493e5d81b9ed/Std/Data/List/Lemmas.lean#L94-L96

This PR makes it an alias.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by dwrensha** [comment]
cc @digama0, who suggested that I should make a PR about this.

This duplication was found by [tryTacticAtEachStep](https://github.com/dwrensha/compfiles/blob/28828353b229aa1ef1fdcaa3be2a656f02869ac8/scripts/tryTacticAtEachStep.lean).

**Comment by dwrensha** [comment]
I could additionally add a `@[deprecated]` attribute, if that is desired.

**Comment by Ruben-VandeVelde** [comment]
Yeah, I would also deprecate and remove the uses in mathlib

**Comment by dwrensha** [comment]
> Yeah, I would also deprecate and remove the uses in mathlib

done

File: Mathlib/Data/List/Basic.lean [hunk 1]
```diff
@@ -97,12 +97,8 @@ theorem _root_.Decidable.List.eq_or_ne_mem_of_mem [DecidableEq α]
 lemma mem_pair {a b c : α} : a ∈ [b, c] ↔ a = b ∨ a = c := by
   rw [mem_cons, mem_singleton]
 
-theorem mem_split {a : α} {l : List α} (h : a ∈ l) : ∃ s t : List α, l = s ++ a :: t := by
-  induction' l with b l ih; {cases h}; rcases h with (_ | ⟨_, h⟩)
-  · exact ⟨[], l, rfl⟩
-  · rcases ih h with ⟨s, t, rfl⟩
-    exact ⟨b :: s, t, rfl⟩
-#align list.mem_split List.mem_split
+@[deprecated] alias mem_split := append_of_mem
```
Comment (YaelDillies) [inline Mathlib/Data/List/Basic.lean:100]:
```suggestion
-- 2024-03-23
@[deprecated] alias mem_split := append_of_mem
```

**Review by YaelDillies — APPROVED** [review]
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**